File: codeExamples/week11/spring24_help_reviews-main/client/src/models/Restaurant.js
export class Restaurant {
  constructor(data) {
    // TODO make databaseitem class to inherit from
    this.id = data.id
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.name = data.name
    this.imgUrl = data.imgUrl
    this.description = data.description
    this.visits = data.visits
    this.isShutdown = data.isShutdown
    this.creatorId = data.creatorId
    // TODO make profile class
    this.creator = data.creator
    this.reportCount = data.reportCount
  }
}

File: codeExamples/week11/spring24_help_reviews-main/server/Models/Restaurant.cs
namespace help_reviews.Models;

public class Restaurant : RepoItem<int>
{
  public string Name { get; set; }
  public string ImgUrl { get; set; }
  public string Description { get; set; }
  public int Visits { get; set; }
  public bool? IsShutdown { get; set; } // this property can now be null
  public int ReportCount { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }

}
